File: app/src/main/java/prm392/project/model/Blog.java
package prm392.project.model;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Blog {
    @SerializedName("blogId")
    private int blogId;

    @SerializedName("userId")
    private Integer userId;

    @SerializedName("username")
    private String username;

    @SerializedName("blogTitle")
    private String blogTitle;
    @SerializedName("blogDate")
    private String blogDate; // Changed to String to handle "2025-07-01T00:00:00" and avoid API 26 requirement

    @SerializedName("blogLike")
    private Integer blogLike;
    @SerializedName("firstImage")
    private String firstImage; // Changed to String to handle Base64 from API

    @SerializedName("blogImagesBase64")
    private List<String> blogImages; // Changed to String to handle Base64 from API

    @SerializedName("blogContent")
    private String blogContent;

    @SerializedName("eateryLocationDetail")
    private String eateryLocationDetail;


    // Optional fields that might not be in the response
    private Double blogRate;
    private Integer eateryId;
    private Integer blogStatus;
    private String eateryAddressDetail;
    private Integer foodQualityRate;
    private Integer environmentRate;
    private Integer serviceRate;
    private Integer pricingRate;
    private Integer hygieneRate;
    private String opinion;

    // Constructors
    public Blog() {}

    // Getter and Setter methods
    public int getBlogId() {
        return blogId;
    }

    public void setBlogId(int blogId) {
        this.blogId = blogId;
    }

    public Integer getUserId() {
        return userId;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getBlogTitle() {
        return blogTitle;
    }

    public void setBlogTitle(String blogTitle) {
        this.blogTitle = blogTitle;
    }

    public String getBlogContent() {
        return blogContent;
    }

    public void setBlogContent(String blogContent) {
        this.blogContent = blogContent;
    }

    public String getBlogDate() {
        return blogDate;
    }

    public void setBlogDate(String blogDate) {
        this.blogDate = blogDate;
    }

    public Integer getBlogLike() {
        return blogLike;
    }

    public void setBlogLike(Integer blogLike) {
        this.blogLike = blogLike;
    }

    public String getFirstImage() {
        return firstImage;
    }

    public void setFirstImage(String firstImage) {
        this.firstImage = firstImage;
    }

    public List<String> getBlogImages() {
        return blogImages;
    }

    public void setBlogImages(List<String> blogImages) {
        this.blogImages = blogImages;
    }

    public Double getBlogRate() {
        return blogRate;
    }

    public void setBlogRate(Double blogRate) {
        this.blogRate = blogRate;
    }

    public Integer getEateryId() {
        return eateryId;
    }

    public void setEateryId(Integer eateryId) {
        this.eateryId = eateryId;
    }

    public Integer getBlogStatus() {
        return blogStatus;
    }

    public void setBlogStatus(Integer blogStatus) {
        this.blogStatus = blogStatus;
    }

    public String getEateryLocationDetail() {
        return eateryLocationDetail;
    }

    public void setEateryLocationDetail(String eateryLocationDetail) {
        this.eateryLocationDetail = eateryLocationDetail;
    }

    public String getEateryAddressDetail() {
        return eateryAddressDetail;
    }

    public void setEateryAddressDetail(String eateryAddressDetail) {
        this.eateryAddressDetail = eateryAddressDetail;
    }

    public Integer getFoodQualityRate() {
        return foodQualityRate;
    }

    public void setFoodQualityRate(Integer foodQualityRate) {
        this.foodQualityRate = foodQualityRate;
    }

    public Integer getEnvironmentRate() {
        return environmentRate;
    }

    public void setEnvironmentRate(Integer environmentRate) {
        this.environmentRate = environmentRate;
    }

    public Integer getServiceRate() {
        return serviceRate;
    }

    public void setServiceRate(Integer serviceRate) {
        this.serviceRate = serviceRate;
    }

    public Integer getPricingRate() {
        return pricingRate;
    }

    public void setPricingRate(Integer pricingRate) {
        this.pricingRate = pricingRate;
    }

    public Integer getHygieneRate() {
        return hygieneRate;
    }

    public void setHygieneRate(Integer hygieneRate) {
        this.hygieneRate = hygieneRate;
    }

    public String getOpinion() {
        return opinion;
    }

    public void setOpinion(String opinion) {
        this.opinion = opinion;
    }
}

File: BE/BusinessObjects/Models/Blog.cs
using System;
using System.Collections.Generic;

namespace BusinessObjects.Models;

public partial class Blog
{
    public int BlogId { get; set; }

    public int? UserId { get; set; }

    public string? BlogTitle { get; set; }

    public string? BlogContent { get; set; }

    public DateOnly? BlogDate { get; set; }

    public byte[]? BlogBillImage { get; set; }

    public double? BlogRate { get; set; }

    public int? BlogLike { get; set; }

    public int? EateryId { get; set; }

    public int? BlogStatus { get; set; }

    public string? EateryLocationDetail { get; set; }

    public string? EateryNameDetail { get; set; }

    public string? EateryAddressDetail { get; set; }

    public int? FoodQualityRate { get; set; }

    public int? EnvironmentRate { get; set; }

    public int? ServiceRate { get; set; }

    public int? PricingRate { get; set; }

    public int? HygieneRate { get; set; }

    public string? Opinion { get; set; }

    public DateTime? PaidExpirationDate { get; set; }

    public virtual ICollection<BlogFoodType> BlogFoodTypes { get; set; } = new List<BlogFoodType>();

    public virtual ICollection<BlogImage> BlogImages { get; set; } = new List<BlogImage>();

    public virtual ICollection<BlogLike> BlogLikes { get; set; } = new List<BlogLike>();

    public virtual ICollection<BlogMealType> BlogMealTypes { get; set; } = new List<BlogMealType>();

    public virtual ICollection<BlogPriceRange> BlogPriceRanges { get; set; } = new List<BlogPriceRange>();

    public virtual ICollection<Bookmark> Bookmarks { get; set; } = new List<Bookmark>();

    public virtual ICollection<Comment> Comments { get; set; } = new List<Comment>();

    public virtual ICollection<Report> Reports { get; set; } = new List<Report>();

    public virtual User? User { get; set; }
}
